### PR TITLE
resource/alicloud_cs_kubernetes_addon: Support to modify the custom configuration of the kubernetes cluster addon

### PR DESCRIPTION
**Pull request by saberrey**
1. datasource/alicloud_cs_kubernetes_addon_metadata: Add new data source alicloud_cs_kubernetes_addon_metadata.
2. datasource/alicloud_cs_kubernetes_addons: Add new return field config.
3. resource/alicloud_cs_kubernetes_addon: Support to modify the custom configuration of the kubernetes cluster addon.

### CODE REVIEW

File: website/docs/d/cs_kubernetes_addon_metadata.html.markdown [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Argument Reference
+
+The following arguments are supported.
+* `cluster_id` - (Required) The id of kubernetes cluster.
+* `name` - (Required) The name of the cluster addon. You can get a list of available addons that the cluster can install by using data source `alicloud_cs_kubernetes_addons`.
+* `version` - (Required) The version of the cluster addon.
+
+## Attributes Reference
+
+* `cluster_id` - The id of kubernetes cluster.
```
[the remark this answers is not in the document]
Comment (saberrey) [reply website/docs/d/cs_kubernetes_addon_metadata.html.markdown]: done

File: website/docs/d/cs_kubernetes_addons.html.markdown [hunk 2]
```diff
@@ -40,4 +41,5 @@ The following arguments are supported.
   * `name` - The name of addon. 
   * `current_version` - The current version of addon, if this field is an empty string, it means that the addon is not installed.
   * `next_version` - The next version of this addon can be upgraded to.
-  * `required` - Whether the addon is a system addon.
\ No newline at end of file
+  * `required` - Whether the addon is a system addon.
+  * `current_config` - The current custom configuration of the addon.
```
[the remark this answers is not in the document]
Comment (saberrey) [reply website/docs/d/cs_kubernetes_addons.html.markdown]: done

File: website/docs/r/cs_kubernetes_addon.html.markdown [hunk 3]
```diff
@@ -86,6 +108,7 @@ The following arguments are supported:
 * `cluster_id` - (Required, ForceNew) The id of kubernetes cluster.
 * `name` - (Required, ForceNew) The name of addon.
 * `version` - (Required) The current version of addon.
+* `config` - (Optional) The custom configuration of addon. You can checkout the customizable configuration of the addon through datasource `alicloud_cs_kubernetes_addon_metadata`, the returned format is the standard json schema. If return empty, it means that the addon does not support custom configuration yet. You can also checkout the current custom configuration through the data source `alicloud_cs_kubernetes_addons`.
```
[the remark this answers is not in the document]
Comment (saberrey) [reply website/docs/r/cs_kubernetes_addon.html.markdown]: done

File: alicloud/resource_alicloud_cs_kubernetes_addon.go [hunk 4]
```diff
@@ -136,20 +141,24 @@ func resourceAlicloudCSKubernetesAddonUpdate(d *schema.ResourceData, meta interf
...
+	if d.HasChange("version") {
 		err := csClient.upgradeAddon(d)
 		if err != nil {
 			return WrapErrorf(err, DefaultErrorMsg, ResourceAlicloudCSKubernetesAddon, "upgradeAddon", err)
 		}
-
-		stateConf := BuildStateConf([]string{"running", "Upgrading", "Pause"}, []string{"Success"}, d.Timeout(schema.TimeoutUpdate), 10*time.Second, csClient.CsKubernetesAddonStateRefreshFunc(clusterId, name, []string{"Failed", "Canceled"}))
-		if _, err := stateConf.WaitForState(); err != nil {
-			return WrapErrorf(err, DefaultErrorMsg, ResourceAlicloudCSKubernetesAddon, "WaitForSuccessAfterUpdate", d.Id())
+	} else if d.HasChange("config") {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cs_kubernetes_addon.go:152]: 这个地方不应该是 else 吧，如果 version 和 config 同时修改了，怎么办

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

**Pull request merged**